dbms/src/Storages/KVStore/FFI/FileEncryption.h

```diff
@@ -92,6 +92,11 @@ struct FileEncryptionInfo : private FileEncryptionInfoRaw
     // FileEncryptionRes::Disabled means encryption feature has never been enabled, so no file will be encrypted.
     bool isEncrypted() const { return (res != FileEncryptionRes::Disabled && method != EncryptionMethod::Plaintext); }
 
+    bool operator==(const FileEncryptionInfo & rhs) const
+    {
+        return res == rhs.res && method == rhs.method && *key == *rhs.key && *iv == *rhs.iv;
```

Comment:
better check whether `key` and `iv` is nullptr

Reply:
Fixed https://github.com/pingcap/tiflash/pull/8627/commits/c9df90dc886c8b22796141d81583177c4fac0f78

I change to `bool equals(const FileEncryptionInfo & rhs) const` to avoid accidentally used in another place.